d2l-visible-on-ancestor-behavior.html

```diff
@@ -0,0 +1,119 @@
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="d2l-dom.html">
+<link rel="import" href="../d2l-fastdom-import/fastdom.html">
+
+<script>
+	window.D2L = window.D2L || {};
+	window.D2L.PolymerBehaviors = window.D2L.PolymerBehaviors || {};
+
+	/** @polymerBehavior */
+	D2L.PolymerBehaviors.VisibleOnAncestorBehavior = {
+
+		properties: {
+
+			/**
+			 * Whether the element is only visible when hovered or focus is within ancestor with 'd2l-visible-on-ancestor-target'.
+			 */
+			visibleOnAncestor: {
+				type: Boolean,
+				reflectToAttribute: true
+			}
+
+		},
+
+		attached: function() {
+
+			if (!this.visibleOnAncestor) return;
+			this.__voaTarget = D2L.Dom.findComposedAncestor(this, function(node) {
+				if (!node || node.nodeType !== 1) return false;
+				return (node.classList.contains('d2l-visible-on-ancestor-target'));
+			});
+			if (!this.__voaTarget) return;
+
+			this.__voaHandleBlur = this.__voaHandleBlur.bind(this);
+			this.__voaHandleFocus = this.__voaHandleFocus.bind(this);
+			this.__voaHandleMouseEnter = this.__voaHandleMouseEnter.bind(this);
+			this.__voaHandleMouseLeave = this.__voaHandleMouseLeave.bind(this);
+			this.__voaHandleHideEnd = this.__voaHandleHideEnd.bind(this);
+
+			this.__voaTarget.addEventListener('focus', this.__voaHandleFocus, true);
```

Comment:
Using capture here on `focus` and `blur` handlers so we can run our handler for focus changes within the target.